### PR TITLE
IGNITE-17738 Fix partitions lwm on cluster crush

### PR DESCRIPTION
**Pull request by Mmuzaf**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyCountersTest.java [hunk 1]
```diff
@@ -214,6 +166,8 @@ public BlockableFileIOFactory(FileIOFactory factory) {
      */
     @Test
     public void testCountersOnCrachRecovery() throws Exception {
+        Assume.assumeFalse(atomicityMode == TRANSACTIONAL);
```
Comment (anton-vinogradov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyCountersTest.java:169]: Assume true Atomic?  :)

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyCountersTest.java [hunk 2]
```diff
@@ -154,47 +147,6 @@ public static Iterable<Object[]> data() {
...
-         */
-        public BlockableFileIOFactory(FileIOFactory factory) {
-            this.factory = factory;
-        }
-
-        /** {@inheritDoc} */
-        @Override public FileIO create(File file, OpenOption... modes) throws IOException {
-            return new FileIODecorator(factory.create(file, modes)) {
-                @Override public int write(ByteBuffer srcBuf) throws IOException {
-                    if (ioBlocked)
```
Comment (anton-vinogradov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyCountersTest.java:175]: ioBlocked field has no usages for now

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    protected final ListeningTestLogger listeningLog = new ListeningTestLogger(log);
+
+    /** Number of cluster nodes. */
+    public int nodes = 3;
+
+    /** Number of backups for the default cache. */
+    public int backupNodes = nodes - 1;
+
+    /** */
+    @Before public void beforeEachTest() throws Exception {
```
Comment (anton-vinogradov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java:59]: `Annotation @Before must be on the separated line` (hint from IDEA)

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            .setReadFromBackup(true)); // Allows to check values on backups for idle_verify.
+
+        return cfg;
+    }
+
+    /** */
+    @Test
+    public void testLwmCountersOnClusterCrash() throws Exception {
+        int prepareRqNum = 20;
+        int finishRqNum = 30;
```
Comment (anton-vinogradov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java:91]: Why not just rn? :) 
Let it be shorten to a `Req`, at least %)

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+        for (int i = 0; i < prepareRqNum; i++)
+            GridTestUtils.runAsync(() -> primCache.put(updateCnt.incrementAndGet(), 0));
+
+        prepareLatch.await(getTestTimeout(), TimeUnit.MILLISECONDS);
+
+        for (int i = 0; i < finishRqNum; i++)
+            GridTestUtils.runAsync(() -> primCache.put(updateCnt.incrementAndGet(), 0));
+
+        finishLatch.await(getTestTimeout(), TimeUnit.MILLISECONDS);
```
Comment (anton-vinogradov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java:137]: You never check the result of waiting

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            node -> ((ThrowableFileIOTestFactory)node.configuration().getDataStorageConfiguration().getFileIOFactory())
+                .setThrowEx(true));
+
+        stopAllGrids();
+        startGrids(nodes);
+
+        awaitPartitionMapExchange();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "idle_verify"));
+        assertConflicts(false, false);
```
Comment (anton-vinogradov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java:149]: This code will be successful when all data missed too :) 
It also will be successful at full rebalance (current case) :(

Please check the result properly.

**Review by anton-vinogradov — CHANGES_REQUESTED** [review]

File: modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        for (int i = 0; i < finishRqNum; i++)
+            GridTestUtils.runAsync(() -> primCache.put(updateCnt.incrementAndGet(), 0));
+
+        finishLatch.await(getTestTimeout(), TimeUnit.MILLISECONDS);
+
+        G.allGrids().forEach(
+            node -> ((ThrowableFileIOTestFactory)node.configuration().getDataStorageConfiguration().getFileIOFactory())
+                .setThrowEx(true));
+
+        stopAllGrids();
```
Comment (anton-vinogradov) [inline modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerConsistencyOnClusterCrashTest.java:143]: Counters should be measured before the cluster restart to make sure what corner case this test solves.
